### PR TITLE
Add back end time to Gong

### PR DESCRIPTION
**Pull request by Weves**

### CODE REVIEW

File: backend/danswer/connectors/gong/connector.py [hunk 1]
```diff
@@ -280,9 +289,12 @@ def poll_source(
         # so adding a 1 day buffer and fetching by default till current time
         start_one_day_offset = start_datetime - timedelta(days=1)
         start_time = start_one_day_offset.isoformat()
+
+        # set end_time to `None` if in the `end` is <= 24 hours ago (e.g.
+        # fetch all available calls). Otherwise, respect the passed in `end` time
         end_time = (
             datetime.fromtimestamp(end, tz=timezone.utc).isoformat()
-            if self.use_end_time
+            if end_datetime < datetime.now(tz=timezone.utc) - timedelta(days=1)
```
Comment (yuhongsun96) [inline backend/danswer/connectors/gong/connector.py:297]: I think we don't want this right? We want the latest calls if they're processed. If not then... whatever. I can't remember why I even didn't include end time to begin with

Comment (Weves) [inline backend/danswer/connectors/gong/connector.py]: just in case 🤷

File: backend/danswer/connectors/gong/connector.py [hunk 2]
```diff
@@ -272,6 +277,10 @@ def poll_source(
         else:
             special_start_datetime = datetime.fromtimestamp(0, tz=timezone.utc)
 
+        # don't let the special spart dt be past the end time, this causes issues when
```
Comment (yuhongsun96) [inline backend/danswer/connectors/gong/connector.py]: spart >:( this confused me for longer than I want to admit

File: backend/danswer/connectors/gong/connector.py (hunk 1 above)
Comment (Weves) [reply backend/danswer/connectors/gong/connector.py]: but we can also remove it and just always use the end time - also fine with me

File: backend/danswer/connectors/gong/connector.py [hunk 3]
```diff
@@ -280,9 +289,12 @@ def poll_source(
         # so adding a 1 day buffer and fetching by default till current time
         start_one_day_offset = start_datetime - timedelta(days=1)
         start_time = start_one_day_offset.isoformat()
+
+        # set end_time to `None` if in the `end` is <= 24 hours ago (e.g.
```
Comment (yuhongsun96) [inline backend/danswer/connectors/gong/connector.py]: Why is this a thing specific for Gong?
we could just do it across the board (but maybe 24 hours is a bit long?)

**Pull request merged**